src/Nancy/DynamicDictionary.cs

```diff
@@ -343,5 +347,27 @@ private static string GetNeutralKey(string key)
 
             return data;
         }
+
+        private string DebuggerDisplay
+        {
+            get
+            {
+                var sb = new StringBuilder();
+                sb.Append("{");
+                var maxItens = Math.Min(this.dictionary.Count, 5);
+                for (var i = 0; i < maxItens; i++)
+                {
+                    var item = this.dictionary.ElementAt(i);
+                    sb.AppendFormat(" {0} = {1}{2}", item.Key, item.Value, i < maxItens - 1 ? "," : String.Empty);

```

Comment:
ElementAt returns a KeyValuePair struct, I can wrap in a try block catchin IndexOutOfRange exception, ok?